src/main/java/org/broadinstitute/hellbender/utils/pileup/ReadPileup.java

```diff
@@ -316,6 +320,62 @@ static void fixPairOverlappingQualities(final PileupElement firstElement,
         }
     }
 
+    /**
+     * Returns a new ReadPileup where only one read from an overlapping read
+     * pair is retained.  If the two reads in question disagree to their basecall,
+     * neither read is retained.  If they agree on the base, the read with the higher
+     * base quality observation is retained
+     *
+     * @return the newly filtered pileup
+     */
+    public ReadPileup getOverlappingFragmentFilteredPileup() {
+        return getOverlappingFragmentFilteredPileup(true, true);
+    }
+
+    /**
+     * Returns a new ReadPileup where only one read from an overlapping read
+     * pair is retained.  If discardDiscordant and the two reads in question disagree to their basecall,
+     * neither read is retained.  Otherwise, the read with the higher
+     * quality (base or mapping, depending on baseQualNotMapQual) observation is retained
+     *
+     * @return the newly filtered pileup
+     */
+    public ReadPileup getOverlappingFragmentFilteredPileup(boolean discardDiscordant, boolean baseQualNotMapQual) {
```

Comment:
This needs a unit test in `ReadPileupUnitTest`, especially since it's a bit complicated and has a number of different potential cases.  (likewise, it's no argument overload should probably have a unit test too).